tests/lib/rules/prefer-find-by.test.ts

```diff
@@ -60,35 +76,117 @@ ruleTester.run(RULE_NAME, rule, {
+    ...WAIT_METHODS.map((waitMethod: string) => ({
+      code: `
+        const { getByText, queryByLabelText, findAllByRole } = customRender()
+        it('foo', async () => {
+          const submitButton = await ${waitMethod}(() => getByText('baz', { name: 'button' }))
+        })
+      `,
+      errors: [{
+        messageId: 'preferFindBy',
+        data: {
+          queryVariant: 'findBy',
+          queryMethod: 'Text',
+          fullQuery: `${waitMethod}(() => getByText('baz', { name: 'button' }))`,
+        }
+      }],
+      output: `
+        const { getByText, queryByLabelText, findAllByRole, findByText } = customRender()
+        it('foo', async () => {
+          const submitButton = await findByText('baz', { name: 'button' })
+        })
+      `
+    })),
+    // // this scenario verifies when findBy* were already defined (because it was used elsewhere)
+    ...WAIT_METHODS.map((waitMethod: string) => ({
+      code: `
+        const { getAllByRole, findAllByRole } = customRender()
+        describe('some scenario', () => {
+          it('foo', async () => {
+            const submitButton = await ${waitMethod}(() => getAllByRole('baz', { name: 'button' }))
+          })
+        })
+      `,
+      errors: [{
+        messageId: 'preferFindBy',
+        data: {
+          queryVariant: 'findAllBy',
+          queryMethod: 'Role',
+          fullQuery: `${waitMethod}(() => getAllByRole('baz', { name: 'button' }))`,
+        }
+      }],
+      output: `
+        const { getAllByRole, findAllByRole } = customRender()
+        describe('some scenario', () => {
+          it('foo', async () => {
+            const submitButton = await findAllByRole('baz', { name: 'button' })
+          })
+        })
+      `
+    })),
+    // invalid code, as we need findBy* to be defined somewhere, but required for getting 100% coverage
+    {
+      code: `const submitButton = await waitFor(() => getByText('baz', { name: 'button' }))`,
+      errors: [{
+        messageId: 'preferFindBy',
+        data: {
+          queryVariant: 'findBy',
+          queryMethod: 'Text',
+          fullQuery: `waitFor(() => getByText('baz', { name: 'button' }))`
+        }
+      }],
+      output: `const submitButton = await findByText('baz', { name: 'button' })`
+    },
+    // this code would be invalid too, as findByRole is not defined anywhere.
+    {
+      code: `
+        const getByRole = render().getByRole
+        const submitButton = await waitFor(() => getByRole('baz', { name: 'button' }))
+      `,
+      errors: [{
+        messageId: 'preferFindBy',
+        data: {
+          queryVariant: 'findBy',
+          queryMethod: 'Role',
+          fullQuery: `waitFor(() => getByRole('baz', { name: 'button' }))`
+        }
+      }],
+      output: `
+        const getByRole = render().getByRole
+        const submitButton = await findByRole('baz', { name: 'button' })
```

Comment:
This would turn runnable code into breaking code, right?
Because `findByRole` isn't defined.

If that's the case I would consider this a bad experience as it could potentially break a lot of code.

Reply:
it would.
but it's was already hapenning before this PR. There are a couple of scenarios where the fix breaks.
This PR fixes one of them. This test you've commented is still unfixed. Not quite sure if we can't fix it, as it seems there are multiple scenarios to fix (like I've mentioned in the PR description)

Maybe we went too far by making this rule autofixable? Not sure.

Reply:
In addition to that, I am not sure how often that would be a realistic scenario. I'd think that devs use destructuring or call the function from screen  or the object returned from the render function. But not quite sure. I'm open to debate it.

I still want to enforce this PR _does not introduce_ this scenario, but rather, fixes one of the many possible scenarios where `findBy*` method is added and might not be defined. the scenarios were there already, and the bug mentioned one of them (the one I've fixed)

Reply:
Got it, thanks for the clarification.
For the record, I'm not against it.

I do agree that most devs will use `screen`, especially since we're recommended it.

Reply:
I hope no one uses Testing Library this way to be honest 😅  I think we are fine by covering all the other cases.